### PR TITLE
Feature branch nesting for git-sync

### PR DESCRIPTION
**Pull request by kevgo**
@charlierudolph @allewun @ricmatsui 

Since I had 8 consecutive hours flying back from Germany 2 days ago, here comes the big one: support for nested feature branches!

Since this is such a massive feature, I am dividing and conquering it in the smallest possible increments that keep GT functional. This change makes `git-sync` work with nested branches. If the parent branch is unknown, it asks the user. It stores the information in the Git config. I added copious developer documentation that should explain how this works in detail.

Btw its amazing how useful TDD was here. After a few drawings on paper, I was able to sketch out at least the gist of which commands this should run in a feature spec upfront. This together with the `@debug-commands` flag was super useful to debug the code until it worked. Good times.  :)

implements #529

### CODE REVIEW

File: features/step_definitions/branch_configuration_steps.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Given(/^Git Town has no branch hierarchy information for "(.*?)"$/) do |branch_names|
+  Kappamaki.from_sentence(branch_names).each do |branch_name|
+    run_shell_command "git config --unset git-town.branches.parent.#{branch_name}"
+    run_shell_command "git config --unset git-town.branches.parents.#{branch_name}"
+  end
+end
+
```
Comment (kevgo) [inline features/step_definitions/branch_configuration_steps.rb]: fixed

**Comment by kevgo** [comment]
FYI I have added verification for the branch information known to Git Town in https://github.com/Originate/git-town/commit/cc0954e438d7ada19752a1605e9eb70ae112214c

**Comment by allewun** [comment]
I'm late to this, but I think "parents" is a bit misleading. With the way you've defined hierarchical branches, branches can only have one parent. "Ancestors" seems to be more fitting, what do you think?

**Comment by kevgo** [comment]
You are right. I'll change it. Good catch!

**Pull request closed without merging**